runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1061,7 +1064,8 @@ private void process(
     if (workItem.getSourceState().getOnlyFinalize()) {
       outputBuilder.setSourceStateUpdates(Windmill.SourceState.newBuilder().setOnlyFinalize(true));
       work.setState(State.COMMIT_QUEUED);
-      workCommitter.commit(Commit.create(outputBuilder.build(), computationState, work));
+      workProcessingContext.queueCommit(
```

Comment:
for example if work/workprocessingcontext were combined we could have a single
work.queueCommit(...) which would internally update it's state

Reply:
done